src/main/java/mate/academy/bookshop/mapper/BookMapper.java

```diff
@@ -5,10 +5,14 @@
 import mate.academy.bookshop.dto.BookRequestDto;
 import mate.academy.bookshop.model.Book;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
+
     BookDto toDto(Book book);
 
+    @Mapping(target = "isbn", expression = "java(mate.academy.bookshop.util.IsbnFormatter"
+            + ".format(bookDto.getIsbn()))")

```

Comment:
transforming the data that was sent by user is usually not the best idea, unless it is an explicit contract between the client and API, in most cases if data is not of a format that we accept - we want to throw an exception

Reply:
we do not change the data, we bring it to a single form for storage in the database, you may know similar examples with the phone in the registration form, you can enter the number in different forms, for example: 0990000000 || +380990000000 || +380 99 000 00 00 || +380 99-000-00-00 and so on, but even on the UI the site will format your number as needed

Reply:
Got it! I'd personally prefer to just set a strict format of the ISBN's that we accept and throw an exception on attempt to save the ISBN in an invalid format, but I think this approach can also work :)